File: EDUARDO/Matlab_OO_RT/setup_function_rt_descartar.m
% Classe que descarta os L primeiros dados
classdef setup_function_rt_descartar < handle
   properties
        a
        I
        L
   end
   methods
       %Construtor inializador do objeto
       % inival: inicia todas as variaveis
        function obj =setup_function_rt_descartar(L)
        
            obj.a=[0 0 0];
            obj.I=0;
            obj.L = L;
            
        end
        
        %ar: input signal  [1x3]
        %return :::
        %a: Sinal retornado [1x3]
        function a = descartar(obj,ar)
            
            if obj.I < obj.L+1
                
                obj.I = obj.I + 1;          
                
            end
            
            if obj.I > obj.L
               
                obj.a = ar;
                
            end
        
            a = obj.a;
            
          end

   end
end

File: EDUARDO/APPs/AppCompleto_v22_NovoDetec/app/src/main/java/fun/eduardo/funcoes/setup_function_rt_descartar.java
package fun.eduardo.funcoes;

import net.trucomanx.pdsplibj.pdsra.*;

public class setup_function_rt_descartar {


	public PdsMatrix a;


	private int i,L;


	public setup_function_rt_descartar(int L){

	
		this.a = new PdsMatrix(1,3);

		this.i = 0;

		this.L = L;	


	}


	//ar: input signal  [1x3]
    //return :::
    //a: Sinal retornado [1x3]

	public PdsMatrix descartar(double ar[]) {

       		      
		if (this.i < this.L)
		{
       
        	this.i = this.i + 1;                
                          
                
         }
            
         if(this.i >= this.L)
         {      
            this.a.SetValue(0,0,ar[0]);
			this.a.SetValue(0,1,ar[1]);
			this.a.SetValue(0,2,ar[2]);
                
         }

	
        return(a);

	}

	
}

